py3status/modules/do_not_disturb.py

```diff
@@ -1,104 +1,225 @@
-        self.running = Event()
+class Dunst(Notification):
+    def toggle(self, state):
+        system('killall --signal {} dunst'.format(self.parent.signals[state]))
+
+
+class Xfce4_notifyd(Notification):
+    def setup(self, parent):
+        from dbus import Interface, SessionBus
+        self.iface = Interface(
+            SessionBus().get_object('org.xfce.Xfconf', '/org/xfce/Xfconf'),
+            'org.xfce.Xfconf',
+        )
+
+    def get_state(self):
+        return self.iface.GetProperty('xfce4-notifyd', '/do-not-disturb')
+
+    def toggle(self, state):
+        self.iface.SetProperty('xfce4-notifyd', '/do-not-disturb', state)
+
+
+class Lxqt_notificationd(Notification):
+    def setup(self, parent):
+        from threading import Thread, Event
+        from time import sleep
+        self.process = None
         self.killed = Event()
+        self.running = Event()
+        self.command = 'killall {}'.format(self.server)
 
-        class MyThread(Thread):
+        class LxqtThread(Thread):
             def run(this):
                 while not self.killed.is_set():
                     if self.running.wait():
-                        system("killall '{}'".format(self.notification_manager))
-                        sleep(self.refresh_interval)
+                        system(self.command)
+                        sleep(0.25)
+
+        LxqtThread().start()
+
+    def get_state(self):
+        try:
+            self.parent.py3.command_output(['pgrep', '-fx', self.server])
+            return False
+        except self.parent.py3.CommandError:
+            return True
+
+    def toggle(self, state):
+        if state:
+            self.running.set()
+            self.process_cleanup()
+        else:
+            from subprocess import Popen
+            self.running.clear()
+            self.process = Popen(self.server)
 
-        MyThread().start()
+    def kill(self):
+        self.killed.set()
+        self.running.set()
+        self.process_cleanup()
+
+    def process_cleanup(self):
+        if self.process:
+            self.process.kill()
+            self.process = None
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 10
+    format = '{server} [\?if=state&color=good DND|\?color=bad DND]'
+    server = None
+    signals = ('SIGTERM', 'SIGUSR1')
+    state = 'last'
+
+    class Meta:
+        deprecated = {
+            'rename': [
+                {
+                    'param': 'notification_manager',
+                    'new': 'server',
+                    'msg': 'obsolete parameter use `server`',
+                },
+            ],
+        }
+
+    def post_config_hook(self):
+        # deprecate dunst_signals
+        if self.signals == ('SIGTERM', 'SIGUSR1'):
+            self.signals = (
+                getattr(self, 'dunst_signal_off', self.signals[0]),
+                getattr(self, 'dunst_signal_on', self.signals[1])
+            )
```

Comment:
If deprecated why not documented as such or even remove this properly?

Reply:
You were not here and I had trouble removing this properly. :stuck_out_tongue_winking_eye: 

BAD CODE AHEAD.

```python
def post_config_hook(self):
    # (lazy) deprecate dunst_signals
    if self.signals == ('SIGTERM', 'SIGUSR1'):
        self.signals = (
            getattr(self, 'dunst_signal_off', self.signals[0]),
            getattr(self, 'dunst_signal_on', self.signals[1])
        )
    # (lazy) deprecate state_off, state_on
    on = getattr(self, 'state_on', None)
    off = getattr(self, 'state_off', None)
    if on is None and off is None:
        on = off = 'DND'
    else:
        if not on:
            on = 'ON'
        if not off:
            off = 'OFF'
    # (lazy) deprecate {state} in format
    self.format = getattr(self, 'format', None)
    if self.format:
        fmt = self.format
    else:
        fmt = '[\?if=state&color=good {}|\?color=bad {}]'.format(on, off)
```